### DISCUSSION TITLE
Retrieving chain id(network id) from JAVA SCORE

### DISCUSSION THREAD

**Discussion by ajaya-man** [icon-project/btp#62] [Q&A]
Is it possible to get the chain id inside the JAVA SCOREs similar to how its done in solidity?

https://ethereum.stackexchange.com/questions/56749/retrieve-chain-id-of-the-executing-chain-from-a-solidity-contract

**Comment by ICONationDevTeam** [comment]
It looks like it is available [from the ServiceManager](https://github.com/icon-project/goloop/blob/d4569e3c0e5f38489a8400b3e7dc7a7c01da2172/module/service.go#L337) but I haven't seen anything to get that info from Java Score

I'm not sure if that method is publicly exposed to avm through IPC proxy, it'll need some core devs opinion to know if that's easily implementable

**Comment by sink772** [comment]
@ajaya-man Why do you need this interface in SCORE level?  We need a justification for adding the new interface.
